samza-core/src/main/java/org/apache/samza/container/grouper/task/GroupByContainerCount.java

```diff
@@ -211,7 +220,7 @@ private void validateTasks(Set<TaskModel> tasks) {
   private List<TaskGroup> createContainers(int startContainerId, int endContainerId) {
     List<TaskGroup> containers = new ArrayList<>(endContainerId - startContainerId);
     for (int i = startContainerId; i < endContainerId; i++) {
```

Comment:
nit: seems like this is the reason why this Grouper does not work w/ non-integer processorIds? We probably should open a JIRA to fix it.